### PR TITLE
Custom Hex-String Encoder

### PR DESCRIPTION
**Pull request by mikiquantum**

### CODE REVIEW

File: proofs/tree.go [hunk 1]
```diff
@@ -772,65 +791,6 @@ func normalizeDottedProperty(prop string) string {
 	return prop
 }
 
-// getDottedValueByProperty takes a dotted property and retrieves the interface value of an object
```
Comment (mikiquantum) [inline proofs/tree.go]: Unused code

File: proofs/tree.go [hunk 2]
```diff
@@ -138,21 +137,29 @@ import (
...
 const DefaultSaltsLengthSuffix = "Length"
 
+type defaultValueEncoder struct{}
+
+func (valueEncoder *defaultValueEncoder) encodeToString(value []byte) string {
+	return hexutil.Encode(value)
+}
+
+type ValueEncoder interface {
+	encodeToString([]byte) string
```
Comment (vedhavyas) [inline proofs/tree.go]: `encodeToString` must be exported else no one can implement the interface outside the package

File: proofs/tree.go [hunk 3]
```diff
@@ -177,12 +185,13 @@ type DocumentTree struct {
...
 	hash              hash.Hash
 	saltsLengthSuffix string
+	valueEncoder      ValueEncoder
 }
 
 func (doctree *DocumentTree) String() string {
 	return fmt.Sprintf(
 		"DocumentTree with Hash [%s] and [%d] leaves",
-		hex.EncodeToString(doctree.RootHash()),
+		doctree.valueEncoder.encodeToString(doctree.RootHash()),
```
Comment (vedhavyas) [inline proofs/tree.go]: This will panic if the valueEncoder is nil

**Comment by mikiquantum** [comment]
@vedhavyas thanks for the comments, just addressed them.

**Pull request merged**